src/main/java/wl/user/entity/RoleEntity.java

```diff
@@ -0,0 +1,62 @@
+package wl.user.entity;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import java.util.Objects;
+import lombok.Getter;
+import lombok.Setter;
+import org.hibernate.annotations.Cache;
+import org.hibernate.annotations.CacheConcurrencyStrategy;
+import org.hibernate.annotations.NaturalId;
+import org.hibernate.annotations.NaturalIdCache;
+
+/**
+ * User entity record.
+ */
+@Entity
+@Table(name = "roles")
+@Cache(
+    usage = CacheConcurrencyStrategy.READ_WRITE
+)
+@NaturalIdCache
+@Getter
+@Setter
+public final class RoleEntity implements wl.contract.Entity<Long> {
+
```

Comment:
Changes related to comment above (`.map(RoleEntity::new)`)

```suggestion
public final class RoleEntity implements wl.contract.Entity<Long> {

/**
 * Constructor with a parameter to initialize the role by name.
 */
public RoleEntity(String name) {
    this.name = name;
}

/**
 * Default constructor required by JPA.
 */
public RoleEntity() {}
```